socialhome/streams/app/tests/stores/streamStore.tests.js

```diff
@@ -0,0 +1,95 @@
+import "chai/register-should"
+import {Server, WebSocket} from "mock-socket"
+
+import Vuex from "vuex"
+
+import {newinstance, stateOperations} from "streams/app/stores/streamStore"
+import {actions, mutations} from "streams/app/stores/streamStore.operations"
+
+
+describe("streamStore", () => {
+    afterEach(() => {
+        Sinon.restore()
+    })
+
+    describe("newinstance", () => {
+        context("when websocket connects", () => {
+            it("should connect with protocol wss:// when browser protocol is HTTPS", () => {
+                jsdom.reconfigure({url: "https://localhost"})
+                window.context = {streamName: "public"}
+                const mockWebSocket = Sinon.spy()
+                newinstance(mockWebSocket)
+                mockWebSocket.getCall(0).args[0].should.equal("wss://localhost/ch/streams/public/")
+            })
+
+            it("should connect with protocol ws:// when browser protocol is HTTP", () => {
+                jsdom.reconfigure({url: "http://localhost"})
+                window.context = {streamName: "public"}
+                const mockWebSocket = Sinon.spy()
+                newinstance(mockWebSocket)
+                mockWebSocket.getCall(0).args[0].should.equal("ws://localhost/ch/streams/public/")
+            })
+        })
+
+        context("when websocket receives a message", () => {
+            it("should dispatch receivedNewContent to store when serveur sends a nex message", (done) => {
+                let mockserver = new Server("wss://localhost:8080/ch/streams/public/")
+                jsdom.reconfigure({url: "https://localhost:8080"})
+                window.context = {streamName: "public"}
+                Sinon.spy(Vuex.Store.prototype, "dispatch")
+                mockserver.on("connection", () => mockserver.send(JSON.stringify({event: "new", id: 4})))
+                newinstance(WebSocket)
+                setTimeout(() => {
```

Comment:
Mocha has a 2s timeout anyway. We must change that. Up to 10s, I'd say. I made a quick search if Mocha can be programmatically configured in `setup.js` yesterday, but I haven't found yet.